### PR TITLE
Misc fixes to the localemod module

### PR DESCRIPTION
**Pull request by asomers**
### What does this PR do?
1) Fixes the error message for locale.get_locale on unsupported platforms.  Previously it would raise a KeyError
2) Skips the localemod tests on FreeBSD, where that module is unsupported
3) Fixes some localemod tests whose pass conditions were too generous.

### What issues does this PR fix or reference?
none

### Previous Behavior
Calling localemod.get_locale on an unsupported platform would raise an unhelpful KeyError

### New Behavior
Calling localemod.get_locale on an unsupported platform would raise a CommandExecutionError, as it should

### Tests written?
No.  The existing localemod tests are now skipped on FreeBSD.  If any new errors pop up in CI, it's probably because I fixed test_gen_locale and test_set_locale.  Previously, those tests weren't actually asserting anything

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by dwoz** [comment]
@asomers looks like the test failures are related to the change.

**Comment by asomers** [comment]
Yeah.  I'll set up a Linux development environment so I can debug those.

File: tests/integration/modules/test_localemod.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def test_gen_locale(self):
         locale = self.run_function('locale.get_locale')
         new_locale = _find_new_locale(locale)
         ret = self.run_function('locale.gen_locale', [new_locale])
-        self.assertTrue(ret)
```
Comment (s0undt3ch) [inline tests/integration/modules/test_localemod.py]: We should still assert against `ret`, we're testing against `gen_locale`.
Probably `get_locale` again using the new locale, and then assert against the error not being present.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by asomers** [comment]
The failure on ci/py3/centos7/proxy was due to bundler, not my changes.

File: tests/integration/modules/test_localemod.py (hunk 1 above)
Comment (asomers) [reply tests/integration/modules/test_localemod.py]: asserting against `ret` is useless.  It always succeeds, even if the command failed.

Comment (s0undt3ch) [reply tests/integration/modules/test_localemod.py]: Hmm, that suggests a bad implementation.
Not saying it should be addressed in this PR, at least for now.

**Comment by asomers** [comment]
I believe I fixed the failures on CentOS and Fedora

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**